### PR TITLE
Add rst-directive-colons hook

### LINKED ISSUE #32: Check for incorrect sphinx directives
Hi,

In [pandas](https://github.com/pandas-dev/pandas/blob/master/.pre-commit-config.yaml), there's a local hook which checks for instances of, e.g.
```
    .. warning:
```
instead of
```
   .. warning::
```

:
```yaml
    -   id: incorrect-sphinx-directives
        name: Check for incorrect Sphinx directives
        language: pygrep
        entry: |
            (?x)
            # Check for cases of e.g. .. warning: instead of .. warning::
            \.\.\ (
                autosummary|contents|currentmodule|deprecated|
                function|image|important|include|ipython|literalinclude|
                math|module|note|raw|seealso|toctree|versionadded|
                versionchanged|warning
            ):[^:]
        files: \.(py|pyx|rst)$
```

It's not specific to pandas, and others might find it useful - would a PR to include it in pygrep hooks be welcome?

### PR DESCRIPTION
**Pull request by MarcoGorelli**
closes #32

### CODE REVIEW

File: .pre-commit-hooks.yaml [hunk 1]
```diff
@@ -65,3 +65,9 @@
     entry: "\uFFFD"
     language: pygrep
     types: [text]
+-   id: rst-directive-colons
```
Comment (asottile) [inline .pre-commit-hooks.yaml:68]: these should be alphabetically sorted -- can you add a test for this and fix it? (if not, I can add a test -- just fixing it is fine)

File: .pre-commit-hooks.yaml [hunk 2]
```diff
@@ -65,3 +65,9 @@
...
+    name: rst directives end with two colons
+    description: 'Detect mistake of rst directive not ending with double colon'
+    entry: '^\s*\.\. [a-z]+(:|:{3,}|:\s:)$'
```
Comment (asottile) [inline .pre-commit-hooks.yaml:71]: why not `:\s\s:`? if we're going to add more beyond the base case probably use something like `:(?!:).*$`

File: tests/hooks_test.py [hunk 3]
```diff
@@ -209,3 +209,28 @@ def test_text_unicode_replacement_char_positive(s):
...
+
+
+@pytest.mark.parametrize(
+    's',
+    (
+        '    .. warning:',
+        '.. warning:',
+        '.. warning: :',
+        '.. warning:::',
+        '.. warning::::',
```
Comment (asottile) [inline tests/hooks_test.py:221]: are these mistakes people actually make? I wouldn't bother with these and just stick to the original one-colon-is-an-error

Comment (MarcoGorelli) [reply tests/hooks_test.py]: OK, sure, have gone back to the original regular expression you suggested

File: .pre-commit-hooks.yaml (hunk 1 above)
Comment (MarcoGorelli) [reply .pre-commit-hooks.yaml]: I've fixed it here, I can add a test for this in a separate PR if you like, have opened #34 to keep track of this

**Review by asottile — APPROVED** [review]

**Pull request merged**